src/frequenz/sdk/timeseries/_resampling.py

```diff
@@ -653,17 +652,23 @@ def resample(self, timestamp: datetime) -> Sample:
             seconds=period * conf.max_data_age_in_periods
         )
 
-        # We need to pass a dummy Sample to bisect because it only support
-        # specifying a key extraction function in Python 3.10, so we need to
-        # compare samples at the moment.
-        cut_index = bisect(self._buffer, Sample(minimum_relevant_timestamp, None))
+        # Do a linear search to find the min and max indexes for samples
+        # relevant to this window.
+        min_index = 0
+        max_index = len(self._buffer)
+        for i, sample in enumerate(self._buffer):
+            if sample.timestamp <= minimum_relevant_timestamp:
+                min_index = i + 1
+            elif sample.timestamp > timestamp:
+                max_index = i
+                break
```

Comment:
Not sure, but might the builtin function [`MAX()`](https://docs.python.org/3/library/functions.html#max) and `MIN()` be used here?

Reply:
That would need 2 iterations over the array, and I also need the index, not the value. In any case, I might go back to using `bisect` if the performance is better.